tests/test_server.py

```
@@ -0,0 +1,114 @@
+# You would need to add more comprehensive tests
+
+
+class TestGitLabMCP(unittest.TestCase):
+    """Test cases for GitLab MCP server"""
+
+    def setUp(self):
+        """Set up test fixtures"""
+        self.mock_ctx = MagicMock()
+        self.mock_lifespan_context = MagicMock()
+        self.mock_ctx.request_context.lifespan_context = self.mock_lifespan_context
+        self.mock_lifespan_context.token = "fake_token"
```

Ruff (0.15.2)

[error] Possible hardcoded password assigned to: "token"

(S105)